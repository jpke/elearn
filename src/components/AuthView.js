import React from 'react';
import CourseListCreator from '../utils/CourseListCreator'
import CourseList from '../components/CourseList';
import Register from '../components/Register';
import Login from '../components/Login';

//view for authContainer
//will call login or register views when user is not logged in
//will call list of courses available to user when user is logged in
const AuthView = (props) => {
  let courses;
  if(props.courses) {
    courses = CourseListCreator(props.courses, props.selectCourse, props.passed, props.token, props.url);
  }
  return (
    <div className="authContainer">

      {props.token ?
        //token present, indicating user is logged in
        //display courses accessible to user
        <div className="course-list-container">
          <CourseList
            courses={courses}
            course={props.course}
            logOut={props.logOut}
          />
        </div>
      :
        //token is not present, indicating user is not logged in
        //toggle between login and registration views
        props.viewLogin === true ?
        <div className="authFormContainer">
          <h2>Welcome</h2>
          <h4>You must login or </h4>
          <button className="toggleLogin" onClick={props.toggleView}>Register</button>
          <Login logIn={props.logIn} />
        </div>
        :
        <div className="authFormContainer">
          <h2>Welcome</h2>
          <h4>You must register or </h4>
          <button className="toggleLogin" onClick={props.toggleView}>Login</button>
          <Register register={props.register} />
        </div>
        }
    </div>
  );
}

export default AuthView;
