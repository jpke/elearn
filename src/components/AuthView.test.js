/* eslint-disable no-undef */
/* eslint-disable import/no-named-as-default */

import React from 'react';
import renderer from 'react-test-renderer';

import AuthView from './AuthView';

it('renders correctly when logged out', () => {
  const tree = renderer.create(
    <AuthView
      toggleView={jest.fn()}
      register={jest.fn()}
      logIn={jest.fn()}
      viewLogin={false}
      selectCourse={jest.fn()}
      logOut={jest.fn()}
      courses={[{
        name: "example course",
        id: "00000",
        enrollable: [{email: "new@email.com"}],
        quizzes: [
          {
            _id: "1111",
            title: "example Quiz"
          }
        ],
        admin: false
      }]}
      course={[{
        name: "example course",
        id: "00000",
        enrollable: [{email: "new@email.com"}],
        quizzes: [
          {
            _id: "1111",
            title: "example Quiz"
          }
        ],
        admin: false
      }]}
      passed={[{
        _id: "2222",
        of: "1111"
      }]}
      token={""}
      url={"urlAddress"}
      demo={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly when logged in not as admin with no course selected', () => {
  const tree = renderer.create(
    <AuthView
      toggleView={jest.fn()}
      register={jest.fn()}
      logIn={jest.fn()}
      viewLogin={false}
      selectCourse={jest.fn()}
      logOut={jest.fn()}
      courses={[{
        name: "example course",
        id: "00000",
        enrollable: [{email: "new@email.com"}],
        quizzes: [
          {
            _id: "1111",
            title: "example Quiz"
          }
        ],
        admin: false
      }]}
      course={""}
      passed={[{
        _id: "2222",
        of: "1111"
      }]}
      token={"000000"}
      url={"urlAddress"}
      demo={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly when logged in not as admin with course selected', () => {
  const tree = renderer.create(
    <AuthView
      toggleView={jest.fn()}
      register={jest.fn()}
      logIn={jest.fn()}
      viewLogin={false}
      selectCourse={jest.fn()}
      logOut={jest.fn()}
      courses={[{
        name: "example course",
        id: "00000",
        enrollable: [{email: "new@email.com"}],
        quizzes: [
          {
            _id: "1111",
            title: "example Quiz"
          }
        ],
        admin: false
      }]}
      course={[{
        name: "example course",
        id: "00000",
        enrollable: [{email: "new@email.com"}],
        quizzes: [
          {
            _id: "1111",
            title: "example Quiz"
          }
        ],
        admin: false
      }]}
      passed={[{
        _id: "2222",
        of: "1111"
      }]}
      token={"000000"}
      url={"urlAddress"}
      demo={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly when logged in as admin with no course selected', () => {
  const tree = renderer.create(
    <AuthView
      toggleView={jest.fn()}
      register={jest.fn()}
      logIn={jest.fn()}
      viewLogin={false}
      selectCourse={jest.fn()}
      logOut={jest.fn()}
      courses={[{
        name: "example course",
        id: "00000",
        enrollable: [{email: "new@email.com"}],
        quizzes: [
          {
            _id: "1111",
            title: "example Quiz"
          }
        ],
        admin: false
      }]}
      course={""}
      passed={[{
        _id: "2222",
        of: "1111"
      }]}
      token={"000000"}
      url={"urlAddress"}
      demo={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly when logged in as admin with course selected', () => {
  const tree = renderer.create(
    <AuthView
      toggleView={jest.fn()}
      register={jest.fn()}
      logIn={jest.fn()}
      viewLogin={false}
      selectCourse={jest.fn()}
      logOut={jest.fn()}
      courses={[{
        name: "example course",
        id: "00000",
        enrollable: [{email: "new@email.com"}],
        quizzes: [
          {
            _id: "1111",
            title: "example Quiz"
          }
        ],
        admin: false
      }]}
      course={[{
        name: "example course",
        id: "00000",
        enrollable: [{email: "new@email.com"}],
        quizzes: [
          {
            _id: "1111",
            title: "example Quiz"
          }
        ],
        admin: true
      }]}
      passed={[{
        _id: "2222",
        of: "1111"
      }]}
      token={"000000"}
      url={"urlAddress"}
      demo={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
