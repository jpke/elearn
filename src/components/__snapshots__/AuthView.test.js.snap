exports[`test renders correctly when logged in as admin with course selected 1`] = `
<div
  className="authContainer">
  <div
    className="course-list-container">
    <div
      className="course-list">
      <h2 />
      <ul>
        <li
          onClick={[Function]}>
          example course
          <a
            href="urlAddressusers/certificate/example course/000000"
            target="_blank">
            <svg
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 24 24">
              <g>
                <path
                  d="M9 16.17l-4.17-4.17-1.42 1.41 5.59 5.59 12-12-1.41-1.41z" />
              </g>
            </svg>
          </a>
        </li>
      </ul>
      <div>
        <p>
          Choose 
          <a
            className="redirect"
            onClick={[Function]}
            style={Object {}}>
            Quiz
          </a>
           or 
          <a
            className="redirect"
            onClick={[Function]}
            style={Object {}}>
            Lessons
          </a>
           to access course content
        </p>
      </div>
      <button
        onClick={[Function]}>
        Logout
      </button>
    </div>
  </div>
</div>
`;

exports[`test renders correctly when logged in as admin with no course selected 1`] = `
<div
  className="authContainer">
  <div
    className="course-list-container">
    <div
      className="course-list">
      <h2>
        Select course
      </h2>
      <ul>
        <li
          onClick={[Function]}>
          example course
          <a
            href="urlAddressusers/certificate/example course/000000"
            target="_blank">
            <svg
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 24 24">
              <g>
                <path
                  d="M9 16.17l-4.17-4.17-1.42 1.41 5.59 5.59 12-12-1.41-1.41z" />
              </g>
            </svg>
          </a>
        </li>
      </ul>
      
      <button
        onClick={[Function]}>
        Logout
      </button>
    </div>
  </div>
</div>
`;

exports[`test renders correctly when logged in not as admin with course selected 1`] = `
<div
  className="authContainer">
  <div
    className="course-list-container">
    <div
      className="course-list">
      <h2 />
      <ul>
        <li
          onClick={[Function]}>
          example course
          <a
            href="urlAddressusers/certificate/example course/000000"
            target="_blank">
            <svg
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 24 24">
              <g>
                <path
                  d="M9 16.17l-4.17-4.17-1.42 1.41 5.59 5.59 12-12-1.41-1.41z" />
              </g>
            </svg>
          </a>
        </li>
      </ul>
      <div>
        <p>
          Choose 
          <a
            className="redirect"
            onClick={[Function]}
            style={Object {}}>
            Quiz
          </a>
           or 
          <a
            className="redirect"
            onClick={[Function]}
            style={Object {}}>
            Lessons
          </a>
           to access course content
        </p>
      </div>
      <button
        onClick={[Function]}>
        Logout
      </button>
    </div>
  </div>
</div>
`;

exports[`test renders correctly when logged in not as admin with no course selected 1`] = `
<div
  className="authContainer">
  <div
    className="course-list-container">
    <div
      className="course-list">
      <h2>
        Select course
      </h2>
      <ul>
        <li
          onClick={[Function]}>
          example course
          <a
            href="urlAddressusers/certificate/example course/000000"
            target="_blank">
            <svg
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 24 24">
              <g>
                <path
                  d="M9 16.17l-4.17-4.17-1.42 1.41 5.59 5.59 12-12-1.41-1.41z" />
              </g>
            </svg>
          </a>
        </li>
      </ul>
      
      <button
        onClick={[Function]}>
        Logout
      </button>
    </div>
  </div>
</div>
`;

exports[`test renders correctly when logged out 1`] = `
<div
  className="authContainer">
  <div
    className="authFormContainer">
    <h2>
      Welcome
    </h2>
    <h4>
      You must register or 
    </h4>
    <button
      className="toggleLogin"
      onClick={[Function]}>
      Login
    </button>
    <div
      className="registerContainer">
      <h3
        className="title">
        Register
      </h3>
      <form
        id="authForm"
        onSubmit={[Function]}>
        <input
          className="auth"
          id="userName"
          placeholder="name"
          type="text" />
        <input
          className="auth"
          id="email"
          placeholder="email"
          type="email" />
        <input
          className="auth"
          id="password"
          placeholder="password"
          type="password" />
        <button
          className="authButton"
          type="submit">
          Register
        </button>
      </form>
    </div>
  </div>
</div>
`;
